model_config_examples.txt
{
    "data": {
        "window_size": 1,
        "start_offset": 1,
        "columns": [
            "HR", "O2Sat", "Temp", "SBP", "MAP", "DBP", "Resp", "EtCO2",
            "BaseExcess", "HCO3", "FiO2", "pH", "PaCO2", "SaO2", "AST", "BUN",
            "Alkalinephos", "Calcium", "Chloride", "Creatinine",
            "Bilirubin_direct", "Glucose", "Lactate", "Magnesium", "Phosphate",
            "Potassium", "Bilirubin_total", "TroponinI", "Hct", "Hgb", "PTT",
            "WBC", "Fibrinogen", "Platelets", "Age", "Unit1", "Unit2", "Gender",
            "HospAdmTime", "ICULOS"
        ]
    },
    "model": {
        "type": "WeibullCox"
    }
}

{
    "data": {
        "window_size": 6,
        "start_offset": 6,
        "horizon": 6,
        "columns": [
            "HR", "O2Sat", "Temp", "SBP", "MAP", "DBP", "Resp", "EtCO2",
            "BaseExcess", "HCO3", "FiO2", "pH", "PaCO2", "SaO2", "AST", "BUN",
            "Alkalinephos", "Calcium", "Chloride", "Creatinine",
            "Bilirubin_direct", "Glucose", "Lactate", "Magnesium", "Phosphate",
            "Potassium", "Bilirubin_total", "TroponinI", "Hct", "Hgb", "PTT",
            "WBC", "Fibrinogen", "Platelets", "Age", "Unit1", "Unit2", "Gender",
            "HospAdmTime", "ICULOS"
        ]
    },
    "model": {
        "type": "Log",
        "input_shape": [6, 37]
    }
}

{
    "data": {
        "window_size": 6,
        "start_offset": 6,
        "horizon": 6,
        "columns": [
            "HR", "O2Sat", "Temp", "SBP", "MAP", "DBP", "Resp", "EtCO2",
            "BaseExcess", "HCO3", "FiO2", "pH", "PaCO2", "SaO2", "AST", "BUN",
            "Alkalinephos", "Calcium", "Chloride", "Creatinine",
            "Bilirubin_direct", "Glucose", "Lactate", "Magnesium", "Phosphate",
            "Potassium", "Bilirubin_total", "TroponinI", "Hct", "Hgb", "PTT",
            "WBC", "Fibrinogen", "Platelets", "Age", "Unit1", "Unit2", "Gender",
            "HospAdmTime", "ICULOS"
        ]
    },
    "model": {
        "type": "MLP",
        "input_shape": [6, 37],
        "layers": [
            {"type": "Flatten"},
            {"type": "Linear", "in_features": 222, "out_features": 256},
            {"type": "ReLU"},
            {"type": "Linear", "in_features": 256, "out_features": 512},
            {"type": "Sigmoid"},
            {"type": "Linear", "in_features": 512, "out_features": 1},
            {"type": "Sigmoid"}
        ]
    }
}

{
    "data": {
        "window_size": 6,
        "start_offset": 6,
        "horizon": 6,
        "columns": [
            "HR", "O2Sat", "Temp", "SBP", "MAP", "DBP", "Resp", "EtCO2",
            "BaseExcess", "HCO3", "FiO2", "pH", "PaCO2", "SaO2", "AST", "BUN",
            "Alkalinephos", "Calcium", "Chloride", "Creatinine",
            "Bilirubin_direct", "Glucose", "Lactate", "Magnesium", "Phosphate",
            "Potassium", "Bilirubin_total", "TroponinI", "Hct", "Hgb", "PTT",
            "WBC", "Fibrinogen", "Platelets", "Age", "Unit1", "Unit2", "Gender",
            "HospAdmTime", "ICULOS"
        ]
    },
    "model": {
        "type": "ResNet",
        "input_shape": [6, 37],
        "input_channels": 1,
        "output_classes": 1,
        "layers": [
            {"type": "Conv2d", "in_channels": 1, "out_channels": 64, "kernel_size": 3, "stride": 1, "padding": 1},
            {"type": "BatchNorm2d", "num_features": 64},
            {"type": "ReLU"},
            {"type": "ResidualBlockGroup", "in_channels": 64, "out_channels": 64, "num_blocks": 2, "stride": 1},
            {"type": "ResidualBlockGroup", "in_channels": 64, "out_channels": 128, "num_blocks": 2, "stride": 2},
            {"type": "ResidualBlockGroup", "in_channels": 128, "out_channels": 256, "num_blocks": 2, "stride": 2},
            {"type": "AvgPool2d", "kernel_size": 2},
            {"type": "Flatten"},
            {"type": "Linear", "in_features": 1280, "out_features": 1},
            {"type": "Sigmoid"}
        ],
        "saved_path": "../models/resnet_trial_4_1.25147_06_15_2024_16_36_03.pth"
    }
}

{
    "data": {
        "window_size": 72,
        "start_offset": 12,
        "horizon": 6,
        "columns": [
            "HR", "O2Sat", "Temp", "SBP", "MAP", "DBP", "Resp", "EtCO2",
            "BaseExcess", "HCO3", "FiO2", "pH", "PaCO2", "SaO2", "AST", "BUN",
            "Alkalinephos", "Calcium", "Chloride", "Creatinine",
            "Bilirubin_direct", "Glucose", "Lactate", "Magnesium", "Phosphate",
            "Potassium", "Bilirubin_total", "TroponinI", "Hct", "Hgb", "PTT",
            "WBC", "Fibrinogen", "Platelets", "Age", "Unit1", "Unit2", "Gender",
            "HospAdmTime", "ICULOS"
        ]
    },
    "model": {
        "input_shape": [72, 40],
        "type": "Transformer",
        "layers": [
            {"type": "Linear", "in_features": 40, "out_features": 512},
            {"type": "PositionalEncoding", "d_model": 512, "seq_len": 72, "dropout": 0.1},
            {"type": "TransformerEncoderLayer", "d_model": 512, "nhead": 8, "dim_feedforward": 2048, "dropout": 0.1, "batch_first": true},
            {"type": "Linear", "in_features": 512, "out_features": 1},
            {"type": "MaxLayer"},
            {"type": "Sigmoid"}
        ],
        "saved_path": "../models/Transformer_rich-snowflake-30_29_0.00062_07_17_2024_01_12_50.pth"
    }
}

{
    "data": {
        "dataset": "MultiTaskDataset",
        "window_size": 6,
        "start_offset": 6,
        "horizon": 6,
        "lead": 0,
        "forecast_len": 6,
        "target": "Hypoxia",
        "columns": [
            "HR", "O2Sat", "SBP", "MAP", "DBP",
            "Resp", "EtCO2", "FiO2", "PaCO2", "SaO2",
            "BaseExcess", "HCO3", "pH", "Lactate", "Hgb", "Temp"
        ]
    },
    "model": {
        "type": "MultiTask",
        "input_shape": [6, 16],
        "modules": {
            "embedding": [
                {"type": "Linear", "in_features": 16, "out_features": 256},
                {"type": "PositionalEncoding", "d_model": 256, "seq_len": 6, "dropout": 0.1}
            ],
            "encoder": [
                {"type": "TransformerEncoderLayer", "d_model": 256, "nhead": 8, "dim_feedforward": 1024, "dropout": 0.1, "batch_first": true}
            ],
            "reconstructor": [
                {"type": "TransformerDecoderLayer", "d_model": 256, "nhead": 8, "dim_feedforward": 1024, "dropout": 0.1, "batch_first": true},
                {"type": "Linear", "in_features": 256, "out_features": 16}
            ],
            "forecaster": [
                {"type": "TransformerDecoderLayer", "d_model": 256, "nhead": 8, "dim_feedforward": 1024, "dropout": 0.1, "batch_first": true},
                {"type": "Linear", "in_features": 256, "out_features": 16}
            ],
            "classifier": [
                {"type": "Linear", "in_features": 256, "out_features": 1},
                {"type": "MaxLayer"},
                {"type": "Sigmoid"}
            ]
        }
    }
}

config.py
SEQ_LEN    = 120
SEQ_OFFSET = 48
BATCH_SIZE = 512
LR         = 0.001
EPOCHS     = 100
INPUT_DIM  = 37
HIDDEN_DIM = 64
FF_DIM     = 1024
N_CLASSES  = 2
N_HEADS    = 16
N_LAYERS   = 8

ORGAN_HIDDEN_DIM = 16
ORGAN_N_HEADS    = 4
ORGAN_N_LAYERS   = 4
ORGAN_FF_DIM     = 128

sepsis_ids_filepath = "../data/sepsis_ids.json"
train_ids_filepath = "../data/train_ids.json"
test_ids_filepath = "../data/test_ids.json"
synthetic_train_ids_filepath = '../data/synthetic_train_ids.json'
synthetic_test_ids_filepath = '../data/synthetic_test_ids.json'
label_dirpath = '../results/labels/'
normals_filepath = "../data/normals.json"

wandb_project = 'hypoxia'

preprocess_method = "standardized_padded"

All_COLS = ['HR', 'O2Sat', 'Temp', 'SBP', 'MAP', 'DBP', 'Resp', 'EtCO2',
       'BaseExcess', 'HCO3', 'FiO2', 'pH', 'PaCO2', 'SaO2', 'AST', 'BUN',
       'Alkalinephos', 'Calcium', 'Chloride', 'Creatinine',
       'Bilirubin_direct', 'Glucose', 'Lactate', 'Magnesium', 'Phosphate',
       'Potassium', 'Bilirubin_total', 'TroponinI', 'Hct', 'Hgb', 'PTT',
       'WBC', 'Fibrinogen', 'Platelets', 'Age', 'Unit1', 'Unit2', 'Gender',
       'HospAdmTime', 'ICULOS']

COL_IDX_MAP = {'HR': 0, 'O2Sat': 1, 'Temp': 2, 'SBP': 3, 'MAP': 4, 'DBP': 5, 'Resp': 6, 'EtCO2': 7, 'BaseExcess': 8, 'HCO3': 9, 'FiO2': 10, 'pH': 11, 'PaCO2': 12, 'SaO2': 13, 'AST': 14, 'BUN': 15, 'Alkalinephos': 16, 'Calcium': 17, 'Chloride': 18, 'Creatinine': 19, 'Bilirubin_direct': 20, 'Glucose': 21, 'Lactate': 22, 'Magnesium': 23, 'Phosphate': 24, 'Potassium': 25, 'Bilirubin_total': 26, 'TroponinI': 27, 'Hct': 28, 'Hgb': 29, 'PTT': 30, 'WBC': 31, 'Fibrinogen': 32, 'Platelets': 33, 'Age': 34, 'Unit1': 35, 'Unit2': 36, 'Gender': 37, 'HospAdmTime': 38, 'ICULOS': 39}


clinical_measurements_dict = {
    "Cardiovascular": ["HR", "SBP", "MAP", "DBP", "TroponinI"],
    "Respiratory": ["O2Sat", "Resp", "EtCO2", "FiO2", "PaCO2", "SaO2", "pH"],
    "Immune": ["Temp", "WBC"],
    "Coagulation": ["Platelets", "PTT", "Fibrinogen"],
    "Renal": ["BUN", "Creatinine", "BaseExcess", "HCO3", "pH", "Chloride", "Magnesium", "Phosphate", "Potassium"],
    "Liver": ["AST", "Alkalinephos", "Bilirubin_direct", "Bilirubin_total"],
    "Hematologic": ["Hct", "Hgb", "PTT", "WBC", "Fibrinogen", "Platelets"],
    "Metabolism": ["BaseExcess", "HCO3", "pH", "Glucose", "Lactate", "Calcium"],
    "Demographic": ["Age", "Gender", "ICULOS"]
}


organ_system_ids = {
    'Cardiovascular': [0, 3, 4, 5, 27],
    'Respiratory': [1, 6, 7, 10, 12, 13, 11],
    'Immune': [2, 31],
    'Coagulation': [33, 30, 32],
    'Renal': [15, 19, 8, 9, 11, 18, 23, 24, 25],
    'Liver': [14, 16, 20, 26],
    'Hematologic': [28, 29, 30, 31, 32, 33],
    'Metabolism': [8, 9, 11, 21, 22, 17],
    "Demographic": [34, 35, 36]
}